ZenPacks/zenoss/WBEM/modeler/wbem.py

```diff
@@ -25,31 +25,54 @@
 set to succesfully pull data.
 
 """
+import itertools
 
 from twisted.internet import ssl, reactor
 from twisted.internet.defer import DeferredList, CancelledError
 
 from Products.DataCollector.plugins.CollectorPlugin import PythonPlugin
 
-from ZenPacks.zenoss.WBEM.utils import addLocalLibPath, result_errmsg
+from ZenPacks.zenoss.WBEM.utils import (
+    addLocalLibPath,
+    result_errmsg,
+    create_connection,
+)
 
 from ZenPacks.zenoss.WBEM.patches import (
     EnumerateClassNames,
     EnumerateClasses,
     EnumerateInstanceNames,
-    EnumerateInstances
+    EnumerateInstances,
+    OpenEnumerateInstances,
+    PullInstances,
 )
 
 addLocalLibPath()
 
 
+def getEnumerateInstances(creds, classname, namespace='root/cimv2', **kwargs):
+    if kwargs.get('MaxObjectCount', 0) > 0:
+        klass = OpenEnumerateInstances
+    elif kwargs.get('OperationTimeout', 0) > 0:
```

Comment:
What you're doing here can be done with 'or' operator:
`if kwargs.get('MaxObjectCount', 0) > 0 or kwargs.get('OperationTimeout', 0) > 0:`
`    klass = OpenEnumerateInstances`